			   READLINE PATCH REPORT
			   =====================

Readline-Release: 6.2
Patch-ID: readline62-003

Bug-Reported-by:	Max Horn <max@quendi.de>
Bug-Reference-ID:	<20CC5C60-07C3-4E41-9817-741E48D407C5@quendi.de>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-readline/2012-06/msg00005.html

Bug-Description:

A change between readline-6.1 and readline-6.2 to prevent the readline input
hook from being called too frequently had the side effect of causing delays
when reading pasted input on systems such as Mac OS X.  This patch fixes
those delays while retaining the readline-6.2 behavior.

Index: readline6-6.2/input.c
===================================================================
--- readline6-6.2.orig/input.c	2010-05-30 22:33:01.000000000 +0000
+++ readline6-6.2/input.c	2012-09-17 12:14:56.863471057 +0000
@@ -409,7 +409,7 @@
 int
 rl_read_key ()
 {
-  int c;
+  int c, r;
 
   rl_key_sequence_length++;
 
@@ -429,14 +429,18 @@
 	{
 	  while (rl_event_hook)
 	    {
-	      if (rl_gather_tyi () < 0)	/* XXX - EIO */
+	      if (rl_get_char (&c) != 0)
+		break;
+		
+	      if ((r = rl_gather_tyi ()) < 0)	/* XXX - EIO */
 		{
 		  rl_done = 1;
 		  return ('\n');
 		}
+	      else if (r == 1)			/* read something */
+		continue;
+
 	      RL_CHECK_SIGNALS ();
-	      if (rl_get_char (&c) != 0)
-		break;
 	      if (rl_done)		/* XXX - experimental */
 		return ('\n');
 	      (*rl_event_hook) ();
Index: readline6-6.2/patchlevel
===================================================================
--- readline6-6.2.orig/patchlevel	2012-09-17 12:14:26.063451590 +0000
+++ readline6-6.2/patchlevel	2012-09-17 12:14:56.863471057 +0000
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-2
+3
